src/IO/CascadeWriteBuffer.h

```diff
@@ -47,6 +47,8 @@ class CascadeWriteBuffer : public WriteBuffer
 
 private:
 
+    void finalizeImpl() override { /* no op */ }
```

Comment:
Shouldn't we finalize buffers from `prepared_sources` here?

Reply:
Will add it.